config/management/src/key.rs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{error::Error, SecureBackends};
+use libra_crypto::ed25519::Ed25519PublicKey;
+use libra_secure_storage::{Storage, Value};
+use std::convert::TryInto;
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+pub struct AssociationKey {
+    #[structopt(flatten)]
+    pub secure_backends: SecureBackends,
+}
+
+impl AssociationKey {
+    pub fn execute(self) -> Result<Ed25519PublicKey, Error> {
+        submit_key(
+            libra_global_constants::ASSOCIATION_KEY,
+            self.secure_backends,
+        )
+    }
+}
+
+#[derive(Debug, StructOpt)]
+pub struct OperatorKey {
+    #[structopt(flatten)]
+    pub secure_backends: SecureBackends,
+}
+
+impl OperatorKey {
+    pub fn execute(self) -> Result<Ed25519PublicKey, Error> {
+        submit_key(libra_global_constants::OPERATOR_KEY, self.secure_backends)
+    }
+}
+
+#[derive(Debug, StructOpt)]
+pub struct OwnerKey {
+    #[structopt(flatten)]
+    pub secure_backends: SecureBackends,
+}
+
+impl OwnerKey {
+    pub fn execute(self) -> Result<Ed25519PublicKey, Error> {
+        submit_key(libra_global_constants::OWNER_KEY, self.secure_backends)
+    }
+}
+
+fn submit_key(key_name: &str, secure_backends: SecureBackends) -> Result<Ed25519PublicKey, Error> {
+    let local: Box<dyn Storage> = secure_backends.local.try_into()?;
+    if !local.available() {
+        return Err(Error::LocalStorageUnavailable);
+    }
+
+    let key = local
+        .get_public_key(key_name)
+        .map_err(|e| Error::LocalStorageReadError(e.to_string()))?
+        .public_key;
+
+    if let Some(remote) = secure_backends.remote {
```

Comment:
Silly question (and not directly related to this PR), but under what circumstances do we see a remote not being defined, but allowing this method to still be called? The name "submit_key" suggests to me it's going to submit the key (somewhere), but surely if remote doesn't exist, this method should error? Rather than just silently failing? Unless there's a use-case I'm missing here

Reply:
Good question, all values are returned and printed to the screen. So this tool let's the human also perform the operation in case you only have a local secure end point.